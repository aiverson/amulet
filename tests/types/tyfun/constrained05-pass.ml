class c2 't begin
  type f2
end

instance c2 int begin
  type f2 = bool
end

type foo 'a = Foo : c2 'a => f2 'a -> foo 'a
