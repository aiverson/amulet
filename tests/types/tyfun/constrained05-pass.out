c2 : Infer{'i : type}. 'i -> constraint
f2 : Infer{'i : type}. 'i -> type
foo : Infer{'a : type}. 'a -> type
Foo : Infer{'a : type}. Spec{'a : 'a}. c2 'a => f2 'a -> foo 'a
